ci/analyze.sh

```diff
@@ -33,108 +33,43 @@ FLUTTER_DIR="$SRC_DIR/flutter"
-    return 1
-  else
-    echo "Success: no issues found in $last_arg"
-  fi
-  return 0
-)
 
-echo "Analyzing dart:ui library..."
-analyze \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$SRC_DIR/out/host_debug_unopt/gen/dart-pkg/sky_engine/lib/ui/ui.dart"
-
-echo "Analyzing spirv library..."
-analyze \
-  --packages="$FLUTTER_DIR/lib/spirv/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/lib/spirv"
-
-echo "Analyzing ci/"
-analyze \
-  --packages="$FLUTTER_DIR/ci/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/ci"
-
-echo "Analyzing flutter_frontend_server..."
-analyze \
-  --packages="$FLUTTER_DIR/flutter_frontend_server/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/flutter_frontend_server"
-
-echo "Analyzing tools/licenses..."
-analyze \
-  --packages="$FLUTTER_DIR/tools/licenses/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/tools/licenses/analysis_options.yaml" \
-  "$FLUTTER_DIR/tools/licenses"
-
-echo "Analyzing testing/litetest"
-analyze \
-  --packages="$FLUTTER_DIR/testing/litetest/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/litetest"
-
-echo "Analyzing testing/benchmark"
-analyze \
-  --packages="$FLUTTER_DIR/testing/benchmark/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/benchmark"
-
-echo "Analyzing testing/smoke_test_failure"
-analyze \
-  --packages="$FLUTTER_DIR/testing/smoke_test_failure/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/smoke_test_failure"
-
-echo "Analyzing testing/dart..."
-analyze \
-  --packages="$FLUTTER_DIR/testing/dart/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/dart"
-
-echo "Analyzing testing/scenario_app..."
-analyze \
-  --packages="$FLUTTER_DIR/testing/scenario_app/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/scenario_app"
-
-echo "Analyzing testing/symbols..."
-analyze \
-  --packages="$FLUTTER_DIR/testing/symbols/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/symbols"
-
-echo "Analyzing githooks..."
-analyze \
-  --packages="$FLUTTER_DIR/tools/githooks/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/tools/githooks"
-
-echo "Analyzing tools/clang_tidy"
-analyze \
-  --packages="$FLUTTER_DIR/tools/clang_tidy/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/tools/clang_tidy"
+echo "Using dart from $DART_BIN"
+"$DART" --version
+echo ""
+
+# Analyze dart:ui.
+dart analyze "$FLUTTER_DIR/lib/ui"
```

Comment:
The only thing this might be missing is that the previous way validates that `_embedder.yaml` generated is correct, whereas this does not. So we will miss if someone messes up `_embedder.yaml`

Reply:
I believe this was previously just validating `out/host_debug_unopt/gen/dart-pkg/sky_engine/lib/ui/hooks.dart` (and perhaps the part files in `out/host_debug_unopt/gen/dart-pkg/sky_engine/lib/ui/` as a side effect)?

Through it would also be good to validate `_embedder.yaml` for sure.

Reply:
`dart` -> `"$DART"` here and below.

Reply:
This previously validated the whole dart:ui generated library AFAICT. It does not just look at hooks.dart.